runtime/engine/src/test/java/io/aklivity/zilla/runtime/engine/internal/ReconfigureIT.java

```diff
@@ -0,0 +1,139 @@
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.aklivity.zilla.runtime.engine.internal;
+
+import static io.aklivity.zilla.runtime.engine.EngineConfiguration.ENGINE_DRAIN_ON_CLOSE;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.rules.RuleChain.outerRule;
+
+import java.io.File;
+
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.DisableOnDebug;
+import org.junit.rules.TestRule;
+import org.junit.rules.Timeout;
+import org.kaazing.k3po.junit.annotation.Specification;
+import org.kaazing.k3po.junit.rules.K3poRule;
+import org.openjdk.jmh.util.FileUtils;
+
+import io.aklivity.zilla.runtime.engine.test.EngineRule;
+import io.aklivity.zilla.runtime.engine.test.annotation.Configuration;
+
+
+public class ReconfigureIT
+{
+    private final K3poRule k3po = new K3poRule()
+        .addScriptRoot("net", "io/aklivity/zilla/specs/engine/streams/network")
+        .addScriptRoot("app", "io/aklivity/zilla/specs/engine/streams/application");
+
+    private final TestRule timeout = new DisableOnDebug(new Timeout(10, SECONDS));
+
+    private final EngineRule engine = new EngineRule()
+        .directory("target/zilla-itests")
+        .commandBufferCapacity(1024)
+        .responseBufferCapacity(1024)
+        .counterValuesBufferCapacity(8192)
+        .configure(ENGINE_DRAIN_ON_CLOSE, false)
+        .configurationRoot("io/aklivity/zilla/runtime/engine/internal")
+        .external("app0")
+        .external("app1")
+        .clean();
+
+    @Rule
+    public final TestRule chain = outerRule(engine).around(k3po).around(timeout);
+
+    @After
+    public void cleanupFileSystem() throws Exception
+    {
+        String source = new File(ReconfigureIT.class
+            .getResource("zilla.reconfigure.original.json")
+            .toURI()).getAbsolutePath();
+        String target = new File("target/test-classes/" +
+            "io/aklivity/zilla/runtime/engine/internal/zilla.reconfigure.json").getAbsolutePath();
+        FileUtils.copy(source, target);
+        new File("target/test-classes/" +
+            "io/aklivity/zilla/runtime/engine/internal/zilla.reconfigure.missing.json").delete();
+    }
+
+    @Test
+    @Configuration("zilla.reconfigure.json")
+    @Specification({
+        "${app}/client.sent.data.reconfigure.modify/server",
+        "${net}/client.sent.data.reconfigure.modify/client"
+    })
+    public void shouldConnectToZillaOnNewPortWhenConfigModified() throws Exception
```

Comment:
Rename to `shouldReconfigureWhenModified()`.

Reply:
Forgot to rename after moving it from TCP binding. Done